modules/mq-instance/main.tf

```diff
@@ -4,22 +4,36 @@
 # Create an MQ on Cloud SaaS instance
 ##############################################################################
 
+# Validation
+# Approach based on https://stackoverflow.com/a/66682419
+locals {
+  # validate username_password or arbitrary secret has a password payload
+  capacity_validate_condition = var.existing_mq_capacity_crn == null && var.subscription_id == null
+  capacity_validate_msg       = "Exactly one of 'existing_mq_capacity_crn' or 'subscription_id' is required."
+  # tflint-ignore: terraform_unused_declarations
+  capacity_validate_check = regex("^${local.capacity_validate_msg}$", (!local.capacity_validate_condition ? local.capacity_validate_msg : ""))
+}
+
 locals {
   # Capacity instance
   split_capacity_crn        = var.existing_mq_capacity_crn != null ? split(":", var.existing_mq_capacity_crn) : []
   existing_mq_capacity_guid = length(local.split_capacity_crn) >= 8 ? local.split_capacity_crn[7] : null
 
   mq_capacity_guid = var.existing_mq_capacity_crn != null ? local.existing_mq_capacity_guid : ibm_resource_instance.mqcloud_capacity[0].id
+  capacity_plan    = var.subscription_id == null ? "reserved-capacity" : "reserved-capacity-subscription"
 }
 
 resource "ibm_resource_instance" "mqcloud_capacity" {
-  count             = var.existing_mq_capacity_crn != null ? 0 : 1
+  count             = var.existing_mq_capacity_crn == null ? 1 : 0
   location          = var.region
   name              = var.name
-  plan              = "reserved-capacity"
+  plan              = local.capacity_plan
   resource_group_id = var.resource_group_id
-  service           = "mqcloud"
-  tags              = var.tags
+  parameters = {
+    "subscription_id" = var.subscription_id
```

Comment:
maybe it would be better to explicitly set to empty string when null
```
"subscription_id" = var.subscription_id != null ? var.subscription_id : ""
```

another thought coming from ease service: maybe it shouldn't be passed at all when null. Unless the service expects it in any case

Reply:
MQ on Cloud has an ongoing use case where there may be no subscription id.

```hcl
  dynamic "parameters" {
    for_each = var.subscription_id != null ? [1] : []
    content {
      "subscription_id" = var.subscription_id
    }
  }
```

Would be a solution to totally avoid the block which only contains subscription id when it is not presented.

Reply:
On second thoughts it is not needed. One of existing_crn or subscription_id is required. If subscription_id is passed, it used here. If existing_crn is passed this resource is not created.